query/src/test/scala/filodb/query/exec/BinaryJoinSetOperatorSpec.scala

```diff
@@ -1369,5 +1370,227 @@ class BinaryJoinSetOperatorSpec extends AnyFunSpec with Matchers with ScalaFutur
   }
+
+  case class KeyedTupleRangeVector(rvKey: Map[ZeroCopyUTF8String, ZeroCopyUTF8String], values: Seq[(Long, Double)])
+    extends RangeVector {
+
+    import NoCloseCursor._
+
+    def key: RangeVectorKey = CustomRangeVectorKey(rvKey)
+
+    def rows(): RangeVectorCursor = values.map{ case (ts, value) => SeqRowReader(Seq[Any](ts, value))}.iterator
+
+    /**
+     * If Some, then it describes start/step/end of output data.
+     * Present only for time series data that is periodic. If raw data is requested, then None.
+     */
+    def outputRange: Option[RvRange] = None
+
+  }
+
+  private def rangeVectors(keyedTs: List[(Map[ZeroCopyUTF8String, ZeroCopyUTF8String], Seq[(Long, Double)])]): List[RangeVector]
+  = keyedTs.map{case (key, value) => SerializedRangeVector(KeyedTupleRangeVector(key, value), resultSchema.columns)}.toList
+
+
+  it("should return true when isEmpty called on emptyRV") {
+    val exec = SetOperatorExec(QueryContext(), dummyDispatcher, Nil, Nil, BinaryOperator.LUnless, Nil, Nil, "_metric_", None)
+    val emptyRv = KeyedTupleRangeVector(Map.empty, Seq.empty)
+    exec.isEmpty(emptyRv, resultSchema) shouldEqual true
+  }
+
+
+  it("should return true when isEmpty called on rv with all NaNs") {
+    val exec = SetOperatorExec(QueryContext(), dummyDispatcher, Nil, Nil, BinaryOperator.LUnless, Nil, Nil, "_metric_", None)
+    val emptyRv = KeyedTupleRangeVector(Map.empty, Seq((0, Double.NaN), (10, Double.NaN), (20, Double.NaN)))
+    exec.isEmpty(emptyRv, resultSchema) shouldEqual true
+  }
+
+  it("should return false when isEmpty called on rv with at least one non NaN") {
+    val exec = SetOperatorExec(QueryContext(), dummyDispatcher, Nil, Nil, BinaryOperator.LUnless, Nil, Nil, "_metric_", None)
+    val emptyRv = KeyedTupleRangeVector(Map.empty, Seq((0, Double.NaN), (10, 1.0), (20, Double.NaN)))
+    exec.isEmpty(emptyRv, resultSchema) shouldEqual false
+  }
+
+  private def rvRowsToListOfTuples(rv: RangeVector) = {
+    rv.rows().map(x => (x.getLong(0), x.getDouble(1))).toList
+  }
+
+  it("should perform A - B when no on is given correctly") {
+    val exec = SetOperatorExec(QueryContext(), dummyDispatcher, Nil, Nil, BinaryOperator.LUnless, Nil, Nil, "_metric_", None)
+    val lhsRv = rangeVectors(List(
+      (Map( "label1".utf8 -> "value1".utf8)-> Seq((0, Double.NaN), (10, 1.0), (20, Double.NaN))),
+      (Map( "label2".utf8 -> "value2".utf8, "onLabel".utf8 -> "onValue1".utf8)-> Seq((0, 1.0), (10, 2.0), (20, 3.0))),
+      (Map( "label2".utf8 -> "value2".utf8, "onLabel".utf8 -> "onValue1".utf8)-> Seq((10, 1.0), (20, 2.0), (30, 3.0))),
+      (Map( "label1".utf8 -> "value1".utf8, "onLabel".utf8 -> "onValue1".utf8)-> Seq((100, 1.0), (200, 2.0), (300, 3.0)))
+    ))
+    val rhsRv = rangeVectors(List(
+      (Map( "label1".utf8 -> "value1".utf8)-> Seq((0, Double.NaN), (10, 1.0), (20, Double.NaN)))
+    ))
+
+    val map = exec.setOpUnless(lhsRv, rhsRv).map( rv => rv.key.labelValues -> rvRowsToListOfTuples(rv)).toMap
```

Comment:
should assert the size or check for the unexpected label-value pairs

Reply:
Good point, missed that in few cases